### ISSUE TITLE
🐛 Bug Report: Alert API shows incorrect message on multiple alerts

### ISSUE THREAD

**Issue by jescalada** [backstage/backstage#28178]
### 📜 Description

Minor bug with `alertApi`.

When multiple alerts (errors) stack, the alert message says "(X older messages)" even though the message doesn't actually update. In other words, the alert should either say "(X newer messages)" or otherwise update to display the latest message on top rather than the oldest message.

### 👍 Expected behavior

It should either update the topmost alert to show the latest alert or append "(X newer messages)" to the first message shown.

### 👎 Actual Behavior with Screenshots

This is a screenshot demonstrating the bug. Clicking "In Progress" on a completed job shows the "not in pending state" error. However, when trying to fail the job, it adds the text "1 older message" even though the oldest message was the "not in pending state" error message, and the latest message was queued.

![older messages bug recording](https://github.com/user-attachments/assets/2685218f-75ed-4733-ae68-475e78af3c6f)

### 👟 Reproduction steps

I can't think of an example off the top of my head, but anything using `alertApi.post()` should trigger it.

This is a code snippet of what I'm doing in my custom plugin component:

```tsx
const discoveryApi = useApi(discoveryApiRef);
const { fetch } = useApi(fetchApiRef);
const alertApi = useApi(alertApiRef);
const handleJobStatusSwitch = async (jobId: string, status: JobStatus) => {
  const statusToEndpointMap: Record<JobStatus, string> = {
    [JobStatus.PENDING]: 'pending',
    [JobStatus.IN_PROGRESS]: 'start',
    [JobStatus.COMPLETED]: 'complete',
    [JobStatus.FAILED]: 'fail',
  };

  try {
    const url = `${await discoveryApi.getBaseUrl('my-custom')}/jobs/${statusToEndpointMap[status]}/${jobId}`;

    const response = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
    });

    const responseJson = await response.json();
    if (!response.ok) throw new Error(responseJson.message);
    alertApi.post({ message: 'Job status updated successfully!', severity: 'success' });

    setTimeout(() => window.location.reload(), 1000);
  } catch (e: any) {
    alertApi.post({ message: `Failed to update job status: ${e}`, severity: 'error', display: 'transient' });
  }
};
```

### 📃 Provide the context for the Bug.

This bug confuses users regarding the order of error/success messages. By default, the `AlertMessage` `display` property is set to `permanent` making it very noticeable for users as they have to manually close the alert.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
@jescalada yep, either of the options that you mentioned seem good to me. Do you want to raise a PR with a suggestion? :pray:

**Comment by kenilkevadiya** [comment]
@benjdlambert, Can I work on this issue?

**Comment by camilaibs** [comment]
Hey @kenilkevadiya 👋🏻 , let's check if @jescalada is still willing to work on this as well. Maybe you guys can collaborate together. I thank you in advance for your help 🙌🏻 .

**Comment by jescalada** [comment]
Hi @kenilkevadiya, pardon the very late response! Feel free to take a shot. I believe fixing this just requires fixing the label on stacked alerts so that it makes sense.

**Comment by awanlin** [comment]
I think this might have been addressed by this PR: https://github.com/backstage/backstage/pull/29390 which got shipped in the recent `1.38.0` release.

**Comment by jescalada** [comment]
Oh, that's right! I'll close this issue. 👍🏼
